posthog/temporal/tests/batch_exports/test_s3_batch_export_workflow.py

```diff
@@ -171,50 +192,78 @@ async def assert_events_in_s3(
     json_data = [json.loads(line) for line in data.decode("utf-8").split("\n") if line]
     # Pull out the fields we inserted only
 
-    json_data.sort(key=lambda x: (x["event"], x["created_at"]))
-
-    # Remove team_id, _timestamp from events
-    if exclude_events is None:
-        exclude_events = []
-
-    def to_expected_event(event):
-        mapping_functions = {
-            "timestamp": to_isoformat,
-            "inserted_at": to_isoformat,
-            "created_at": to_isoformat,
-        }
-        # These are inserted/returned by the ClickHouse event generators, but we do not export them
-        # or we export them as properties.
-        not_exported = {"team_id", "_timestamp", "set", "set_once", "ip", "site_url", "elements"}
-        expected_event = {
-            k: mapping_functions.get(k, lambda x: x)(v) for k, v in event.items() if k not in not_exported
-        }
-
-        if expected_event["inserted_at"] is None:
-            expected_event["inserted_at"] = (
-                dt.datetime.fromisoformat(event["_timestamp"]).replace(tzinfo=dt.timezone.utc).isoformat()
-            )
-        return expected_event
-
-    expected_events = list(
-        map(
-            to_expected_event,
-            (event for event in events if event["event"] not in exclude_events),
-        )
-    )
+    if batch_export_schema is not None:
+        schema_column_names = [field["alias"] for field in batch_export_schema["fields"]]
+    else:
+        schema_column_names = [field["alias"] for field in s3_default_fields()]
 
-    expected_events.sort(key=lambda x: (x["event"], x["created_at"]))
+    json_columns = ("properties", "person_properties", "set", "set_once")
 
-    # First check one event, the first one, so that we can get a nice diff if
-    # the included data is different.
-    assert json_data[0] == expected_events[0]
-    assert json_data == expected_events
+    expected_records = []
+    for record in iter_records(
+        client=clickhouse_client,
+        team_id=team_id,
+        interval_start=data_interval_start.isoformat(),
+        interval_end=data_interval_end.isoformat(),
+        exclude_events=exclude_events,
+        include_events=include_events,
+        fields=batch_export_schema["fields"] if batch_export_schema is not None else s3_default_fields(),
+        extra_query_parameters=batch_export_schema["values"] if batch_export_schema is not None else None,
+    ):
```

Comment:
With custom schemas we can't get the events written to ClickHouse to compare against, as we would have to map them to the custom schema. Instead, we read the events directly from ClickHouse essentially using the same query (or what should be the same query).

This assumes `iter_records` works correctly.